### ISSUE TITLE
EventLoop of Inspector creates high CPU load

### ISSUE THREAD

**Issue by annulen** [qtwebkit/qtwebkit#90]
When Inspector is waiting for user input, it invokes `EventLoop::cycle()`, which in our case performs `QCoreApplication::processEvents()`. 

Calling code is

```
    EventLoop loop;
    while (!m_doneProcessingDebuggerEvents && !loop.ended())
        loop.cycle();
```

It is busy waiting for m_doneProcessingDebuggerEvents flag change, thus wasting CPU.

Need to check if it affects GTK port as well.

**Comment by Emdek** [comment]
It seems to be reproducible for example here: https://videocardz.com/, closing inspector using built-in close button causes short freeze.
But it appears to happen only after inspecting some point on the page first, closing it without that works instantly.

**Comment by annulen** [comment]
```
#0  0x00007f0c1343e84d in poll () at ../sysdeps/unix/syscall-template.S:81
#1  0x00007f0c0db96f54 in g_main_context_poll (priority=2147483647, n_fds=3, fds=0x22da300, 
    timeout=0, context=0x7f0c000016f0) at gmain.c:4103
#2  g_main_context_iterate (context=context@entry=0x7f0c000016f0, block=block@entry=0, 
    dispatch=dispatch@entry=1, self=<optimized out>) at gmain.c:3803
#3  0x00007f0c0db9706c in g_main_context_iteration (context=0x7f0c000016f0, may_block=0)
    at gmain.c:3869
#4  0x00007f0c14264dfc in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/DependenciesQT/Root/lib/libQt5Core.so.5
#5  0x00007f0c154dd118 in WebCore::PageScriptDebugServer::runEventLoopWhilePaused() ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#6  0x00007f0c16334bf6 in Inspector::ScriptDebugServer::handlePause(JSC::JSGlobalObject*, JSC::Debugger::ReasonForPause) ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#7  0x00007f0c14c55393 in JSC::Debugger::pauseIfNeeded(JSC::ExecState*) ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#8  0x00007f0c14c556c0 in JSC::Debugger::exception(JSC::ExecState*, JSC::JSValue, bool) ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#9  0x00007f0c14c6b7a3 in JSC::Interpreter::notifyDebuggerOfExceptionToBeThrown(JSC::ExecState*, JSC::Exception*) ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#10 0x00007f0c14b90a77 in JSC::VM::throwException(JSC::ExecState*, JSC::JSValue) ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#11 0x00007f0c14a578d0 in JSC::JSValue::synthesizePrototype(JSC::ExecState*) const ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#12 0x00007f0c14f229e2 in llint_slow_path_get_by_id ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#13 0x00007f0c14ce90f1 in llint_entry ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
#14 0x00007f0c14cec1eb in llint_entry ()
   from /mnt/ssd2/WebKit/WebKit-new-qt-wk2/WebKitBuild/Release/lib/libQt5WebKit.so.5
```

**Comment by annulen** [comment]
Original issues fixed in 5bdf244d499e5998cd2caeb365fc60c17b878e2d

There is another issues with high JS-related CPU load on start up, presumably related to garbage collection in huge volumes
